### DISCUSSION TITLE
Allow access modifier override from more restrictive to less restrictive

### DISCUSSION THREAD

**Discussion by fa1ntl1ght** [dotnet/csharplang#5572] [Language Ideas]
Currently it is not possible to give an overriden method a less restrictive access modifier.
A workaround exists, where you declare a "proxy" method with a different name and make it invoke the original method, but it adds clutter to the code. So, I propose to allow overriding access modifiers from more restrictive to less restrictive:

```cs
class A
{
    protected virtual void Method(){/*code*/}
}

class B : A
{
    public override void Method(){/*code*/}
}
```
Obviously, this wouldn't work if the base method was private.

Note: the code like above is allowed in Kotlin and Java (and probably other languages too).

**Comment by huoyaoyuan** [comment]
If the base method takes a more restrictive type (protected nested type) as parameter/return, then the override can't change the access modifier.

Since we have solved covariant return, and Java/Kotlin solved this issue, I'd expect it to be straight forward.

**Reply by TonyValenti** [reply]
It seems like you should be able
To make a method override with a More accessible scope but not a less accessible one.

**Reply by HaloFour** [reply]
This should be easier as IIRC this requires no runtime changes.  The runtime allows for explicit overrides as it does explicit implementation, allowing the accessibility and even name to be different as long as the signature is compatible.  This would be more a question of lifting the restrictions in the C# language and emitting the appropriate override modifier in the metadata.

**Reply by voroninp** [reply]
@HaloFour , is there an issue for it? I could not find it.

**Reply by ZzZombo** [reply]
Object Pascal allows this and in fact, the VCL in Delphi expects you to selectively expose protected members as you see it fit when subclassing the built-in controls.
